### ISSUE TITLE
Overhaul sbox.game review process

### ISSUE THREAD

**Issue by matekdev** [Facepunch/sbox-issues#6484]
### For?

S&Box

### What can't you do?

These tickets are somewhat similar so I'll link them here
https://github.com/Facepunch/sbox-issues/issues/6106
https://github.com/Facepunch/sbox-issues/issues/6323

In the long term, the reviewing system should work similarly to how the Steam store works. At least, I feel like that is a good example of a good implementation of reviewing.
- You should be able to reply to reviews as the developer
   - The user who posted the review should get a notification letting them know of the response. To let them know to change the review if they feel the issue was fixed.
- The reviewer's play count should be shown in the review
- Add functionality to report reviews in case they are extremely unhinged (i.e. threats of violence, etc.)
- Players should be able to upvote/downvote reviews depending on whether or not they are trollish reviews
![image](https://github.com/user-attachments/assets/0c9a3a97-2fec-45c1-9ed5-670ceb6cd01a)

### How would you like it to work?

The bullet points I outlined at the top.

### What have you tried?

...

**Comment by MrBrax** [comment]
Agreed, quite annoying with reviews just saying "not working" or confusing criticism, trying to figure out what's actually wrong is hard when you can't reply and ask follow up questions. And if you fix the issue in five minutes, the review author won't ever update the bad review because they don't know it's fixed.

**Comment by anthonysharpy** [comment]
Agree with all of the above

It's annoying having some 14 year old chat absolute shit about your project and then you can't even reply. As someone whose self-worth hinges entirely on things anonymous people have said to them over the internet, this is a big deal! 

But back to being serious, one other thing I don't understand is why 2 out of 3 of the review options are negative emojis ("eh" and poop). I would expect to see like one negative, one neutral (maybe 🙂), and then one positive (maybe 😍)

![Image](https://github.com/user-attachments/assets/109bf99b-567f-437e-ae52-c3b667d81b7d)

**Comment by Retroeer** [comment]
I don't think there should be 3 options at all, 👍 and👎 that's all there needs to be. It works for Steam, would work here too, "shows promise" tells me nothing, the reviewer should clearly state if it sucks or not. Much easier to sort by 👍 or 👎 as a player as well
